app/src/main/java/com/darkempire78/opencalculator/MainActivity.kt

```diff
@@ -136,78 +141,100 @@ class MainActivity : AppCompatActivity() {
-            calculation = calculation.replace('÷', '/')
-            calculation = calculation.replace("log", "log10")
+            if (calculation != "") {
+                calculation = calculation.replace('×', '*')
+                calculation = calculation.replace('÷', '/')
+                calculation = calculation.replace("log", "log10")
 
-            // Add ")" which lack
-            var openParentheses = 0
-            var closeParentheses = 0
+                // Add ")" which lack
+                var openParentheses = 0
+                var closeParentheses = 0
 
-            for (i in 0..calculation.length-1) {
-                if (calculation[i] == '(') {
-                    openParentheses += 1
-                }
-                if (calculation[i] == ')') {
-                    closeParentheses += 1
+                for (i in 0..calculation.length - 1) {
+                    if (calculation[i] == '(') {
+                        openParentheses += 1
+                    }
+                    if (calculation[i] == ')') {
+                        closeParentheses += 1
+                    }
                 }
-            }
-            if (closeParentheses < openParentheses) {
-                for (i in 0..openParentheses-closeParentheses-1) {
-                    calculation += ')'
+                if (closeParentheses < openParentheses) {
+                    for (i in 0..openParentheses - closeParentheses - 1) {
+                        calculation += ')'
+                    }
                 }
-            }
 
-            val exp = Expression(calculation)
-            var result = exp.calculate().toString()
-
-            if (result != "NaN" && result != "Infinity") {
-                // If the double ends with .0 we remove the .0
-                if ((exp.calculate() * 10) % 10 == 0.0) {
-                    result = String.format("%.0f", exp.calculate())
-                    if (result != calculation) {
-                        resultDisplay.setText(result)
+                val exp = Expression(calculation)
+                var result = exp.calculate().toString()
+
+                if (result != "NaN" && result != "Infinity") {
+                    // If the double ends with .0 we remove the .0
+                    if ((exp.calculate() * 10) % 10 == 0.0) {
+                        result = String.format("%.0f", exp.calculate())
+                        if (result != calculation) {
+                            withContext(Dispatchers.Main) {
+                                resultDisplay.setText(result)
+                            }
+                        } else {
+                            withContext(Dispatchers.Main) {
+                                resultDisplay.setText("")
+                            }
+                        }
                     } else {
-                        resultDisplay.setText("")
+                        if (result != calculation) {
```

Comment:
I would personally put the whole if-else blocks in a `withContext` block together.

Reply:
Thank you for your suggestion, I have made these changes as per your suggestion.